fili-core/src/main/java/com/yahoo/bard/webservice/sql/helper/SqlTimeConverter.java

```diff
@@ -0,0 +1,130 @@
+     *
+     * @return the list of sql functions.
+     */
+    List<SqlDatePartFunction> timeGrainToDatePartFunctions(TimeGrain timeGrain);
+
+    /**
+     * Builds the time filters to only select rows that occur within the intervals of the query.
+     * NOTE: you must have one interval to select on.
+     *
+     * @param builder  The RelBuilder used for building queries.
+     * @param intervals  The intervals to select from.
+     * @param timestampColumn  The name of the timestamp column in the database.
+     *
+     * @return the RexNode for filtering to only the given intervals.
+     */
+    RexNode buildTimeFilters(RelBuilder builder, Collection<Interval> intervals, String timestampColumn);
+
+    /**
+     * Gets the number of {@link SqlDatePartFunction} used for the given {@link TimeGrain}.
+     *
+     * @param timeGrain  The timegrain to find groupBy functions for.
+     *
+     * @return the number of functions used to groupBy the given timeGrain.
+     */
+    default int getNumberOfGroupByFunctions(TimeGrain timeGrain) {
+        return timeGrainToDatePartFunctions(timeGrain).size();
+    }
+
+    /**
+     * Builds a list of {@link RexNode} which will effectively groupBy the given {@link TimeGrain}.
+     *
+     * @param builder  The RelBuilder used with calcite to build queries.
+     * @param timeGrain  The timeGrain to build the groupBy for.
+     * @param timeColumn  The name of the timestamp column.
+     *
+     * @return the list of {@link RexNode} needed in the groupBy.
+     */
+    default Stream<RexNode> buildGroupBy(RelBuilder builder, TimeGrain timeGrain, String timeColumn) {
+        return timeGrainToDatePartFunctions(timeGrain)
+                .stream()
+                .map(sqlDatePartFunction -> builder.call(sqlDatePartFunction, builder.field(timeColumn)));
+    }
+
+    /**
+     * Given an array of strings (a row from a {@link java.sql.ResultSet}) and the
+     * {@link TimeGrain} used to make groupBy statements on time, it will parse out a {@link DateTime}
+     * for the row which represents the beginning of the interval it was grouped on.
+     *
+     * @param offset the last column before the date fields.
+     * @param row  The results returned by Sql needed to read the time columns.
+     * @param timeGrain  The timeGrain which was used when calling
+     * {@link #buildGroupBy(RelBuilder, TimeGrain, String)}.
+     *
+     * @return the datetime for the start of the interval.
+     */
+    default DateTime getIntervalStart(int offset, String[] row, TimeGrain timeGrain) {
+        DateTime resultTimeStamp = new DateTime(0, DateTimeZone.UTC);
+
+        List<SqlDatePartFunction> times = timeGrainToDatePartFunctions(timeGrain);
+        for (int i = 0; i < times.size(); i++) {
+            int value = Integer.parseInt(row[offset + i]);
+            SqlDatePartFunction fn = times.get(i);
+            resultTimeStamp = setDateTime(value, fn, resultTimeStamp);
```

Comment:
Not sure if there is a great way to do this with writing out the date as a string and then parsing it (as andrew suggested), but  if there is it would probably be in [ISODateTimeFormat](http://joda-time.sourceforge.net/apidocs/org/joda/time/format/ISODateTimeFormat.html).